### PR TITLE
Feat/refactoring

### PR DESCRIPTION
**Pull request by lesion**
some refactoring addressing recent issues and some bug fix:

- firejail was not working anymore with the addition of --firejail-args (it was threating the argument also if it was empty)
- reuse host's tor cache directory for faster loading / #45 
- add iptables enforcement / #43 
- refactoring error handling
- general cleaning

### CODE REVIEW

File: usr/sbin/orjail [hunk 1]
```diff
@@ -117,33 +118,22 @@ cleanup() {
...
-                 --to-destination $IPHOST:"$TRANSPORT"
-    eno iptables -D OUTPUT -o in-"$NAME" -m state --state ESTABLISHED,RELATED -j ACCEPT
-    eno iptables -D INPUT -i in-"$NAME" -j DROP
-    if [[ $HIDDENSERVICE = y ]]; then
-      eno iptables -D INPUT -i in-"$NAME" -p tcp --destination "$IPNETNS" --dport "$HSERVICEPORT" -j ACCEPT
-      eno iptables -D INPUT -i in-"$NAME" -p tcp --source "$IPNETNS" --sport "$HSERVICEPORT" -j ACCEPT
-    fi
-    eno iptables -D INPUT -i in-"$NAME" -p udp --destination $IPHOST --dport "$DNSPORT" -j ACCEPT
-    eno iptables -D INPUT -i in-"$NAME" -p tcp --destination $IPHOST --dport "$TRANSPORT" -j ACCEPT
+    iptables -S | grep "in-$NAME" | while read -r line; do
```
Comment (bakulf) [inline usr/sbin/orjail]: This is wrong. What about the case where we have in-foo and in-foobar? and $NAME is 'foo' ?

File: usr/sbin/orjail [hunk 2]
```diff
@@ -117,33 +118,22 @@ cleanup() {
...
+      # shellcheck disable=SC2086
+      iptables ${line//-A/-D} &>/dev/null;
+    done
+    iptables -t nat -S | grep "in-$NAME" | while read -r line; do
```
Comment (bakulf) [inline usr/sbin/orjail]: same issue here.

File: usr/sbin/orjail [hunk 3]
```diff
@@ -432,135 +400,136 @@ if ! ip netns list | eno grep -e \\b"$NAME"\\b; then
...
-     echo "DNSPort $IPHOST:$DNSPORT"            >> "$TORCONFIGFILE"
-  else
-     echo "VirtualAddrNetwork $IPNETNS/16"      >> "$TORCONFIGFILE" &&
-     echo "TransListenAddress $IPHOST"          >> "$TORCONFIGFILE" &&
-     echo "TransPort $TRANSPORT"                >> "$TORCONFIGFILE" &&
-     echo "DNSListenAddress $IPHOST"            >> "$TORCONFIGFILE" &&
-     echo "DNSPort $DNSPORT"                    >> "$TORCONFIGFILE"
+  # reuse tor host's cache
+  if [ -d "/var/lib/tor" ]; then
+  mkdir "/tmp/orjail-$NAME"
```
Comment (bakulf) [inline usr/sbin/orjail]: why no error check here and elsewhere?

File: usr/sbin/orjail [hunk 4]
```diff
@@ -220,8 +210,9 @@ fi
 # The tool
 # ~~~~~~~~
 
-# Call cleanup on ctrl-c (SIGINT) and more breakout signals
-trap interrupted HUP INT QUIT TERM ABRT
+# exit on error, and call cleanup on exit
+set -e
```
Comment (bakulf) [inline usr/sbin/orjail]: I like this.

File: usr/sbin/orjail [hunk 5]
```diff
@@ -432,135 +400,136 @@ if ! ip netns list | eno grep -e \\b"$NAME"\\b; then
 
   print G " * Creating a $NAME namespace..."
   # add network namespace
-  ip netns add "$NAME" || die "Failed to add a new namespace"
+  ip netns add "$NAME"
```
Comment (bakulf) [inline usr/sbin/orjail]: set -e is OK but here we don't have the correct error message anymore...

File: usr/sbin/orjail (hunk 1 above)
Comment (lesion) [reply usr/sbin/orjail]: addressed here 4a31fa4eac67c4ac6a75b5b99f9689a085ec8ef7

File: usr/sbin/orjail (hunk 3 above)
Comment (lesion) [reply usr/sbin/orjail]: all errors are catched using `set -e`

File: usr/sbin/orjail (hunk 5 above)
Comment (lesion) [reply usr/sbin/orjail]: solved here e37c7f9118d55b11086a6183b1e26b7972fe6aff

**Pull request merged**